qutebrowser/browser/webengine/notification.py

```diff
@@ -0,0 +1,134 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016-2020 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Handles sending notifications over DBus."""
+
+from typing import Dict, Any
+
+from qutebrowser.utils import log
+
+from PyQt5.QtGui import QImage
+from PyQt5.QtCore import QVariant, QMetaType, QByteArray
+from PyQt5.QtDBus import QDBusConnection, QDBusInterface, QDBus, QDBusArgument
+from PyQt5.QtWebEngineCore import QWebEngineNotification
+from PyQt5.QtWebEngineWidgets import QWebEngineProfile
+
+
+class DBusException(Exception):
+    """Raised when something goes wrong with talking to DBus."""
+
+
+class DBusNotificationManager:
```

Comment:
```suggestion
class DBusNotificationPresenter:
```

given that Qt calls it a presenter as well, and "Manager" is a very generic term (yep, I'm guilty of using it where there would probably be better terms as well!)

Reply:
Naming things is one of the two hard problems, of course :)